### PR TITLE
Add pyAMReX

### PR DESCRIPTION
**Pull request by ax3l**
GPU-Enabled, Zero-Copy AMReX Python Bindings including AI/ML

- [x] depends on #24294

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] ~If static libraries are linked in, the license of the static library is packaged.~
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/pyamrex/bld.bat [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    -DCMAKE_NM=llvm-nm              ^
+    -DCMAKE_VERBOSE_MAKEFILE=ON     ^
+    -DpyAMReX_amrex_internal=OFF    ^
+    -DpyAMReX_pybind11_internal=OFF ^
+    -DPython_EXECUTABLE=%PYTHON%    ^
+    -DPYINSTALLOPTIONS="--no-build-isolation"
+if errorlevel 1 exit 1
+
+:: build
+cmake --build build --config Release --parallel 2
```
Comment (ax3l) [inline recipes/pyamrex/bld.bat:24]:
```suggestion
cmake --build build --config Release --parallel %CPU_COUNT%
```

**Comment by WeiqunZhang** [comment]
Yes, I am willing to be listed as a maintainer.

**Comment by roelof-groenewald** [comment]
I am also willing to be listed as a maintainer. Thanks!

**Comment by atmyers** [comment]
I am also willing to be listed as a maintainer.

**Comment by RemiLehe** [comment]
Yes, I am willing to be listed as a maintainer.

**Comment by ax3l** [comment]
Wat
```
  ERROR (pyamrex,lib/python3.11/site-packages/amrex/space3d/amrex_3d_pybind.cpython-311-darwin.so): Needed DSO lib/libc++.1.dylib found in ['libcxx']
  ERROR (pyamrex,lib/python3.11/site-packages/amrex/space3d/amrex_3d_pybind.cpython-311-darwin.so): .. but ['libcxx'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```

**Comment by ax3l** [comment]
@conda-forge/help-python-c new pybind11 based Python/C++ package ready for review/merge :)

**Comment by ax3l** [comment]
maybe @minrk? :)

**Comment by ax3l** [comment]
@isuruf do you like to review this? :)

**Comment by ax3l** [comment]
@conda-forge/help-python-c new pybind11 based Python/C++ package ready for review/merge :)

**Comment by ax3l** [comment]
@conda-forge/help-python-c Ping: ready for review/merge :)

**Comment by ax3l** [comment]
@conda-forge/help-c-cpp new pybind11 based Python/C++ package ready for review/merge :)

**Comment by dpgrote** [comment]
Yes, I am willing to be listed as a maintainer.

**Comment by ax3l** [comment]
@conda-forge/help-python this is read for review & merge :)

File: recipes/pyamrex/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    - 203.patch
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - clang         # [win]
```
Comment (jaimergp) [inline recipes/pyamrex/meta.yaml:23]: I am not sure we support this. Any reason you can't use the default compilers?

Comment (ax3l) [reply recipes/pyamrex/meta.yaml:23]: Thanks for checking. Yes, this is the recommended way to get modern OpenMP to work on Windows according to @isuruf and works quite well on existing feedstocks.

**Comment by ax3l** [comment]
https://github.com/orgs/conda-forge/teams/help-python-c Ping: ready for review/merge :)

File: recipes/pyamrex/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    - amrex
+  source_files:
+    - tests/*
+  files:
+    - test.sh   # [unix]
+    - test.bat  # [win]
+  commands:
+    # smoke tests: execute
+    - ./test.sh      # [unix and build_platform == target_platform]
+    - call test.bat  # [win]
```
Comment (isuruf) [inline recipes/pyamrex/meta.yaml:61]: If you rename these to `run_test.sh` and `run_test.bat`, they will be run automatically. Not required.

Comment (ax3l) [reply recipes/pyamrex/meta.yaml:61]: Thanks, great hint! Will update in my feedstocks.

File: recipes/pyamrex/meta.yaml (hunk 2 above)
Comment (jaimergp) [reply recipes/pyamrex/meta.yaml:23]: Thanks for the info!

**Pull request merged**